src/main/java/org/beanpod/switchboard/controller/DecoderController.java

```diff
@@ -37,24 +40,30 @@
 
   public static final String UNKNOWN_ERROR_MESSAGE = "Unknown error in DecoderController";
 
+  private final UserDaoImpl userDao;
   private final DecoderDaoImpl decoderDao;
   private final DeviceDaoImpl deviceService;
   private final DecoderMapper decoderMapper;
   private final StreamMapper streamMapper;
   private final DecoderService decoderService;
   private final MaintainDeviceStatus maintainDeviceStatus;
+  private final HttpServletRequest request;
 
   @GetMapping
   public List<DecoderDto> retrieveAllDecoders() {
-    List<DecoderEntity> decoderEntity = decoderDao.getDecoders();
+    UserEntity user = userDao.findUser(request.getUserPrincipal().getName());
+
+    List<DecoderEntity> decoderEntity = decoderDao.getDecoders(user);
     maintainDeviceStatus.maintainStatusField(decoderEntity);
     return decoderMapper.toDecoderDtos(decoderEntity);
   }
 
   @GetMapping("/{serialNumber}")
   public ResponseEntity<DecoderDto> retrieveDecoder(@PathVariable @Valid String serialNumber) {
```

Comment:
This endpoint returns the entire user information as part of the device, including their password hash, we probably don't want that to happen. I think this is because the controller returns a DTO rather than a Model.

Reply:
I'm glad you caught this. It's a side-effect I noticed as well. I wasn't too sure how to fix this. The fact that some controllers used Models and others used DTOs was something I brought up earlier in the project. I was not sure if one was better than the other, so I avoided changing things myself. Thanks to your review comments, we see that Models are the way to go as it allows us to present subset responses. I feel that a conversion effort should be done to go from DTO to Model, and again this should have probably been caught in an earlier review.